catalog/src/main/kotlin/com/adevinta/spark/catalog/examples/ComponentsScreen.kt

```diff
@@ -0,0 +1,136 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.adevinta.spark.catalog.examples
+
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.ExperimentalLayoutApi
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.consumeWindowInsets
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.grid.GridCells
+import androidx.compose.foundation.lazy.grid.GridItemSpan
+import androidx.compose.foundation.lazy.grid.LazyVerticalGrid
+import androidx.compose.foundation.lazy.grid.items
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalLayoutDirection
+import androidx.compose.ui.unit.dp
+import androidx.navigation.NavController
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.rememberNavController
+import com.adevinta.spark.SparkTheme
+import com.adevinta.spark.catalog.examples.component.ComponentItem
+import com.adevinta.spark.catalog.examples.model.Component
+import com.adevinta.spark.components.text.Text
+import com.adevinta.spark.tokens.Layout
+
+@Composable
+public fun ComponentsScreen(
+    modifier: Modifier = Modifier,
+    components: List<Component>,
+    contentPadding: PaddingValues,
+) {
+    val navController = rememberNavController()
+
+    NavHost(
+        modifier = modifier,
+        navController = navController,
+        startDestination = ComponentRoute,
+        builder = {
+            navGraph(
+                navController = navController,
+                contentPadding = contentPadding,
+                components = components,
+            )
+        },
+    )
+}
+
+@OptIn(ExperimentalLayoutApi::class)
+@Composable
+internal fun ComponentsListScreen(
+    components: List<Component>,
+    navController: NavController,
+    contentPadding: PaddingValues,
+) {
+    LazyVerticalGrid(
+        modifier = Modifier
+            .fillMaxSize()
+            .consumeWindowInsets(contentPadding),
+        columns = GridCells.Fixed(Layout.columns / 2),
```

Comment:
nit: this could be a constant. So if we change the number of columns, there is no need to update it in several places individually